pkg/api/customization/cluster/validator.go

```diff
@@ -37,35 +38,49 @@ type Validator struct {
 }
 
 func (v *Validator) Validator(request *types.APIContext, schema *types.Schema, data map[string]interface{}) error {
-	var spec v3.ClusterSpec
+	var clusterSpec v3.ClusterSpec
+	var clientClusterSpec mgmtclient.Cluster
+	logrus.Tracef("Validator: data: %+v\n", data)
+	if err := convert.ToObj(data, &clusterSpec); err != nil {
+		return httperror.WrapAPIError(err, httperror.InvalidBodyContent, "Cluster spec conversion error")
```

Comment:
nit: we'd need different err msg for client and cluster spec

Reply:
Agreed here, error needs to be updated.